### ISSUE TITLE
[Beta voice call] Call quality drops if signal is not open in foreground

### ISSUE THREAD

**Issue by pgera** [WhisperSystems/Signal-Android#6249]
I have:
- [x] searched open and closed issues for duplicates
- [x] read https://github.com/WhisperSystems/Signal-Android/wiki/Submitting-useful-bug-reports

----------------------------------------

### Bug description
Call quality drops if you press the home button, open another app, or send Signal to the background in some other way

### Steps to reproduce
- Initiate Signal call
- Press home button

**Actual result:** Audio starts cutting off
**Expected result:** Shouldn't affect call quality

### Device info

**Device:** Google Pixel Sailfish
**Android version:** 7.1.1 NOF26V
**Signal version:** 3.29.6

### Link to debug log

I have captured the debug log, but the recipient's entire number is visible in plain text after the line CALL_CONNECTED. I'll be happy to send it over a private channel. Also, that seems like a bug in itself

**Comment by mawed** [comment]
@pgera 
> I have captured the debug log, but the recipient's entire number is visible in plain text after the line CALL_CONNECTED. I'll be happy to send it over a private channel. Also, that seems like a bug in itself

As described here: https://github.com/WhisperSystems/Signal-Android/wiki/Submitting-useful-bug-reports

>  Debug log privacy
> 
> Debug logs captured within Signal are stripped from personal information. However please note that the last two digits of your and your contacts phone numbers may be visible in the logs. E.g. +1 555 3455 will show as +******55 in the log. This is to help differentiate the contacts when analysing a log. 

The information will be stripped when uploaded, so this is not a bug.

**Comment by pgera** [comment]
@mawed 
The original gist log that is created doesn't have the information stripped out. It shows the entire number.

In any case, here's the log after I stripped it manually:
http://pastebin.com/6u1F3nvu

**Comment by 2-4601** [comment]
I can't reproduce this log behavior that you are experiencing.
I get:
```
02-20 23:10:35.922 24523 24523 W WebRtcCallActivity: Got message from service: [State: CALL_CONNECTED, recipient: +**********16, identity: null, remoteVideo: false, localVideo: false]
```

If you can reproduce this, could you please file an issue at https://github.com/WhisperSystems/libpastelog/issues. That's the library Signal uses to redact personal info from the logs that are generated via advanced settings.

**Comment by moxie0** [comment]
this is fixed in 3.31.0
